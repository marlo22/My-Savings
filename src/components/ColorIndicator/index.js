import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const StyledColorIndicator = styled.View`
	width: 10;
	height: 40;
	margin-right: 10;
`;

const ColorIndicator = ({ color }) => (
	<StyledColorIndicator style={{ backgroundColor: color }} />
);

ColorIndicator.propTypes = {
	color: PropTypes.string.isRequired
};

export default ColorIndicator;