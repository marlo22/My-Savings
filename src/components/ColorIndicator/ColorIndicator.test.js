import React from 'react';
import { shallow } from 'enzyme';

import ColorIndicator from './';

const color = '#ff0000';

const component = shallow(
  <ColorIndicator color={color} />
);

describe('ColorIndicator', () => {
  test('should render without errors', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should set passed color value', () => {
    const expectedStyle = { backgroundColor: color };
    expect(component.prop('style')).toEqual(expectedStyle);
  });
});