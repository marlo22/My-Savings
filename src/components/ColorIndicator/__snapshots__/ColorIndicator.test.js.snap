// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorIndicator snapshot 1`] = `
<Styled(View)
  style={
    Object {
      "backgroundColor": "#ff0000",
    }
  }
/>
`;
